src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/settings/ProjectSettingsRemoteController.java

```diff
@@ -29,88 +21,16 @@
 @RequestMapping("/projects/{projectId}/settings")
 public class ProjectSettingsRemoteController {
 
-	private MessageSource messageSource;
 	private ProjectService projectService;
-	private ProjectRemoteService projectRemoteService;
-	private UserService userService;
 	private ProjectControllerUtils projectControllerUtils;
 
 	@Autowired
-	public ProjectSettingsRemoteController(MessageSource messageSource, ProjectService projectService,
-			ProjectRemoteService projectRemoteService, UserService userService,
+	public ProjectSettingsRemoteController(ProjectService projectService,
```

Comment:
Do we need 2 separate classes for this?  It's a combined 3 methods between this class and the Ajax controller.

Reply:
@deepsidhu85  and I have been working really hard at keeping all the controllers really lean and only handling what their job is.  A @controller's job is to return a view and that's it, a `@RestController` is to handle an ajax request.  If either of those those actually need some data (which is basically always for the ajax controllers), they can reach out to a specifically designed service.  That way it is easy to know where to go to find/fix a problem.  It keeps the controllers nice and lean and easy to read and way easier to test.